dspace-server-webapp/src/test/java/org/dspace/app/rest/GroupRestRepositoryIT.java

```diff
@@ -1538,4 +1650,342 @@ public void removeMemberUnprocessableTest() throws Exception {
+                GroupBuilder.deleteGroup(parentGroup.getID());
+            }
+        }
+    }
+
+    @Test
+    public void deleteGroupUnprocessableTest() throws Exception {
+        context.turnOffAuthorisationSystem();
+
+        EPerson reviewer1 = EPersonBuilder.createEPerson(context)
+                .withEmail("reviewer1@example.com")
+                .withPassword(password)
+                .build();
+
+        parentCommunity = CommunityBuilder.createCommunity(context)
+                .withName("Parent Community")
+                .build();
+        Community child1 = CommunityBuilder.createSubCommunity(context, parentCommunity)
+                .withName("Sub Community")
+                .build();
+        Collection col1 = CollectionBuilder.createCollection(context, child1)
+                .withName("Collection 1")
+                .withWorkflowGroup(1, admin, reviewer1)
+                .build();
+
+        final Group workflowGroup = col1.getWorkflowStep1(context);
+        context.commit();
+        context.reloadEntity(workflowGroup);
+
+        String authToken = getAuthToken(admin.getEmail(), password);
+
+        getClient(authToken).perform(
+                get("/api/eperson/groups/" + workflowGroup.getID())
+        ).andExpect(status().isOk());
+
+        getClient(authToken).perform(
+                delete("/api/eperson/groups/" + workflowGroup.getID())
+        ).andExpect(status().isUnprocessableEntity());
+
+        getClient(authToken).perform(
+                get("/api/eperson/groups/" + workflowGroup.getID())
+        ).andExpect(status().isOk());
+
+    }
+
+    @Test
+    public void deletePermanentGroupUnprocessableTest() throws Exception {
+        context.turnOffAuthorisationSystem();
+        GroupService groupService = EPersonServiceFactory.getInstance().getGroupService();
+        final Group group = groupService.findByName(context, Group.ANONYMOUS);
+        context.restoreAuthSystemState();
+        String token = getAuthToken(admin.getEmail(), password);
+
+        String authToken = getAuthToken(admin.getEmail(), password);
+
+        getClient(authToken).perform(
+                get("/api/eperson/groups/" + group.getID())
+        ).andExpect(status().isOk());
+
+        getClient(authToken).perform(
+                delete("/api/eperson/groups/" + group.getID())
+        ).andExpect(status().isUnprocessableEntity());
+
+        getClient(authToken).perform(
+                get("/api/eperson/groups/" + group.getID())
+        ).andExpect(status().isOk());
+
+    }
+
+    @Test
+    public void deleteGroupForbiddenTest() throws Exception {
+
+        GroupService groupService = EPersonServiceFactory.getInstance().getGroupService();
+
+        Group parentGroup = null;
+
+        try {
+            context.turnOffAuthorisationSystem();
+
+            parentGroup = groupService.create(context);
+
+            context.commit();
+
+            parentGroup = context.reloadEntity(parentGroup);
```

Comment:
Same as above. should use GroupBuilder, missing a context.restoreAuthSystemState(); and same question about commit() and reloadEntity() as above.